src/pid/CtrePIDController.java

```diff
@@ -0,0 +1,61 @@
+package pid;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.IMotorControllerEnhanced;
+import pid.interfaces.PIDController;
+import sensors.counter.CtreEncoder;
+
+public class CtrePIDController extends CtreController implements PIDController {
+
+  public CtrePIDController(IMotorControllerEnhanced ctreMotorController, CtreEncoder ctreEncoder) {
+    super(ctreMotorController, ctreEncoder);
+  }
+
+  public CtrePIDController(IMotorControllerEnhanced ctreMotorController, CtreEncoder ctreEncoder,
+                           int pidSlot, int timeoutMs) {
+    super(ctreMotorController, ctreEncoder, pidSlot, timeoutMs);
+  }
+
+  public CtrePIDController(IMotorControllerEnhanced ctreMotorController, CtreEncoder ctreEncoder,
+                           double kP, double kI, double kD, double kF, int pidSlot, int timeoutMs) {
+    super(ctreMotorController, ctreEncoder, kP, kI, kD, kF, pidSlot, timeoutMs);
+  }
+
+
+  @Override
+  public double getProcessVariable() {
+    if (this.ctreMotorController.getControlMode() == ControlMode.Velocity) {
+      return this.ctreMotorController.getSelectedSensorVelocity(pidSlot);
+    } else if (this.ctreMotorController.getControlMode() == ControlMode.Position) {
+      return this.ctreMotorController.getSelectedSensorPosition(pidSlot);
+    }
+    return -1;
+  }
+
+  @Override
+  public double getCurrentError() {
+    return this.ctreMotorController.getClosedLoopError(pidSlot);
+  }
+
+  public boolean isOnTarget(double tolerance) {
```

Comment:
`isOnTarget` should move to the `PIDController` interface, and `tolerance` should be fixed in the ctor.

Reply:
But then what about aboveTolerance and belowTolerance in the other isOnTarget? Will that function receive two parameters?

Reply:
Both should move together, and `tolerance` should still be set in ctor.
If you use a builder you can use two `setTolerance` methods, one that accepts a single tolerance and one that accepts both 
`aboveTolerance` and `belowTolerance`.

Also, look at the Markdown Turorial in the bottom right of the comment textbox
![image](https://user-images.githubusercontent.com/10786057/100336088-69d8e780-2fde-11eb-9342-0f8c1d79140d.png)
to see how to make `variables` and [links](https://guides.github.com/features/mastering-markdown/)

Reply:
Talking about tolerance and `isOnTarget`, I now read the previous reviews and saw [this comment I made](https://github.com/OnyxTronix2231/Lib-FRC/pull/18#discussion_r422470682), which is still accurate.
We do need to extract the implementation to a different class, to be able to create others in the future (like the stopping one we usually code-hacked with CTRE by waiting).